Source/WebInspectorUI/UserInterface/Views/ScreenshotsTimelineView.js

```diff
@@ -69,22 +68,42 @@ WI.ScreenshotsTimelineView = class ScreenshotsTimelineView extends WI.TimelineVi
         this.element.removeChildren();
 
         for (let record of this._visibleRecords()) {
-            let imageElement = this.element.appendChild(document.createElement("img"));
-            imageElement.src = record.imageData;
+            this.element.appendChild(this._imageElementForRecord.getOrInitialize(record, () => {
+                let imageElement = document.createElement("img");
+
+                imageElement.hidden = true;
+                imageElement.addEventListener("load", (event) => {
+                    imageElement.hidden = false;
+                }, {once: true});
+                imageElement.src = record.imageData;
 
-            if (record === this._selectedRecord)
-                imageElement.classList.add("selected");
+                imageElement.addEventListener("click", (event) => {
+                    this._selectTimelineRecord(record);
+                });
 
-            imageElement.addEventListener("click", (event) => {
-                this._selectTimelineRecord(record);
-            });
+                return imageElement;
+            }));
         }        
     }
 
     selectRecord(record)
     {
+        if (record === this._selectedRecord)
+            return;
+
+        if (this._selectedRecord)
+            this._imageElementForRecord.get(this._selectedRecord)?.classList.remove("selected");
+
         this._selectedRecord = record;
-        this.needsLayout();
+
+        if (this._selectedRecord) {
+            let element = this._imageElementForRecord.get(this._selectedRecord);
```

Comment:
Same question as in ScreenshotsTimelineOverviewGraph regarding selecting an element we haven't created yet.